std/experimental/allocator/building_blocks/aligned_block_list.d

```diff
@@ -0,0 +1,690 @@
+// Written in the D programming language.
+/**
+Source: $(PHOBOSSRC std/experimental/allocator/building_blocks/_aligned_block_list.d)
+*/
+module std.experimental.allocator.building_blocks.aligned_block_list;
+
+import std.experimental.allocator.common;
+import std.experimental.allocator.building_blocks.null_allocator;
+
+// Common function implementation for thread local and shared AlignedBlockList
+private mixin template AlignedBlockListImpl(bool isShared)
+{
+    import std.traits : hasMember;
+    import std.typecons : Ternary;
+
+    static if (isShared)
+    import core.internal.spinlock : SpinLock;
+
+private:
+    // Doubly linked list of 'AlignedBlockNode'
+    // Each node contains an `Allocator` followed by its payload
+    static struct AlignedBlockNode
+    {
+        AlignedBlockNode* next, prev;
+        Allocator bAlloc;
+
+        static if (isShared)
+        {
+            shared(size_t) bytesUsed;
+            // Since the lock is not taken when allocating, this acts like a refcount
+            // keeping the node alive
+            uint keepAlive;
+        }
+        else
+        {
+            size_t bytesUsed;
+        }
+    }
+
+    // Root of the internal doubly linked list
+    AlignedBlockNode* root;
+
+    // Number of active nodes
+    int numNodes;
+
+    // If the numNodes exceeds this limit, we will start deallocating nodes
+    enum maxNodes = 50;
```

Comment:
I'm curious as to where this number came from. Was it just a gut feeling? Did you make a decision to bias towards "probably not too high" or "probably not too low"?

Reply:
Depending on what the caller passes, `maxNodes` should be different.
The deallocation of the blocks provided by `ParentAllocator` happen lazily inside `allocate`. 
I didn't want to delete nodes too often, but also I didn't want to have high memory usage so I hardcoded something for my tests.
I'll investigate here a bit more to see if I can get rid of this, or maybe pass it as a parameter somehow.
Thanks!